### DISCUSSION TITLE
Velero Restore fails with "Backup not found" (executed from ci/cd pipeline)

### DISCUSSION THREAD

**Discussion by mame-leadec** [vmware-tanzu/velero#8111] [Community support Q&A] [answered]
Im using Velero in a github action ci/cd pipeline to deploy a k8s development cluster. 

Velero is installed with: 

`velero install \
    --provider aws \
    --plugins velero/velero-plugin-for-aws:v1.2.1,velero/velero-plugin-for-csi:v0.7.1\
    --bucket <redacted> \
    --secret-file credentials-velero \
    --use-volume-snapshots=true \
    --use-node-agent \
    --features=EnableCSI \
    --backup-location-config region="eu-central-1",s3ForcePathStyle="false"`
    
immediately after this step a velero restore is executed with:
    
`velero restore create --from-backup cluster-backup`

which fails with:

`An error occurred: backups.velero.io "cluster-backup" not found`

When i check manually at a later point in time, with `velero backup get` i can see that the backup is present.

Does Velero (kopia) need some time to spin up and gather all backup information from S3?

**Comment by kaovilai** [comment, marked as answer]
Yes velero need some time. Would advise your gh actions to wait until backup is available before restore is created.

**Comment by mame-leadec** [comment]
Thank you.
